File: sdkwork-backend-sdk-go/types/plus_ppt_template_form.go
package types

// PPT template creation form
type PlusPptTemplateForm struct {
	Title string `json:"title"`
	Description string `json:"description"`
	CoverImages ImageMediaResourceList `json:"coverImages"`
	Type string `json:"type"`
	Status string `json:"status"`
	AuthorId int `json:"authorId"`
	AuthorName string `json:"authorName"`
	Tags PlusPptTemplateTags `json:"tags"`
	Meta PlusPptTemplateMeta `json:"meta"`
	Slides []PlusPptTemplateSlide `json:"slides"`
}

File: sdkwork-backend-sdk-typescript/src/types/plus-ppt-template-form.ts
import type { ImageMediaResourceList } from './image-media-resource-list';
import type { PlusPptTemplateMeta } from './plus-ppt-template-meta';
import type { PlusPptTemplateSlide } from './plus-ppt-template-slide';
import type { PlusPptTemplateTags } from './plus-ppt-template-tags';

/** PPT template creation form */
export interface PlusPptTemplateForm {
  /** Template title */
  title: string;
  /** Template description */
  description?: string;
  /** Cover images */
  coverImages?: ImageMediaResourceList;
  /** Template type */
  type: 'DEFAULT' | 'BUSINESS' | 'EDUCATION' | 'REPORT' | 'OTHER';
  /** Template status */
  status: 'DEFAULT' | 'ACTIVE' | 'INACTIVE' | 'DRAFT';
  /** Author ID */
  authorId: number;
  /** Author name */
  authorName?: string;
  /** Template tags */
  tags?: PlusPptTemplateTags;
  /** Template metadata */
  meta?: PlusPptTemplateMeta;
  /** Slides */
  slides?: PlusPptTemplateSlide[];
}
